modules/Attendance/Services/TimeConstraintService.php

```
@@ -0,0 +1,702 @@
+        if ($isApproved) {
+            return false;
+        }
+        // Check if overtime exceeds threshold for approval
+        $approvalThresholdMinutes = (int)($config['approval_threshold_minutes'] ?? 0);
+        if ($approvalThresholdMinutes > 0 && $overtimeMinutes <= $approvalThresholdMinutes) {
+            return false; // No approval needed if under threshold
+        }
+        // Return violation details
+        return [
+            'constraint_type' => AttendanceConstraint::TIME_OVERTIME_APPROVAL,
+            'severity' => $this->getSeverityFromConfig($config),
+            'message' => 'Overtime requires approval.',
+            'details' => [
+                'overtime_minutes' => $overtimeMinutes,
+                'approval_threshold' => $approvalThresholdMinutes,
+                'is_approved' => $isApproved
+            ]
+        ];
+    }
+
+    /**
+     * Validates multiple periods constraints.
+     *
+     * This function now performs three main checks in order of priority:
+     * 1. Checks if the attendance date is a specified holiday.
+     * 2. Checks if the day of the week is a scheduled day off (e.g., weekend).
+     * 3. If it's a working day, it ensures the clock-in time is within an allowed work period.
+     *
+     * @param \Modules\Attendance\Models\Attendance $attendance The attendance record to validate.
+     * @param array $config The constraint configuration containing 'weekly_schedule' and 'holidays'.
+     * @return bool|array Returns false if no violation is found, or an array with violation details.
+     */
+    public function validateMultiplePeriods(Attendance $attendance, array $config): bool|array
+    {
+        // If no weekly schedule is defined, we cannot perform this validation.
+        if (!isset($config['weekly_schedule']) || !is_array($config['weekly_schedule'])) {
+            // This isn't a violation, just a misconfiguration. Silently pass.
```

The comment `// This isn't a violation, just a misconfiguration. Silently pass.` is misleading since the function returns `false` which typically indicates a validation failure. Consider either:
1. Returning `true` to indicate validation passed
2. Throwing an exception for misconfiguration
3. Making the comment more accurate about the return value